### PR TITLE
Clear cached builds before new builds

### PR DESCRIPTION
**Pull request by arthurattwell**

### CODE REVIEW

**Review by bertuss — APPROVED** [review]
Looks good to me.

File: test.sh [hunk 1]
```diff
@@ -80,12 +80,27 @@ then
...
 
+    # Clear the cache of any previous builds
+    # https://docs.cloudbees.com/docs/cloudbees-codeship/latest/basic-builds-and-configuration/dependency-cache#_clearing_the_cache
+    # but we use a more direct approach to only delete the contents
+    # of the _site folder, while avoiding the dangerous rm-rf.
+    echo "Deleting previous builds from the cache..."
+    find "$HOME/cache/_site" -mindepth 1 -delete
+
+    # Check if cache is now empty
+    if [ -n "$(ls -A $HOME/cache/_site)" ]; then
```
Comment (bertuss) [inline test.sh:91]: Not specifically applicable to this script, but `[[ ]]` statements are usually preferable to `[ ]` statements:
https://stackoverflow.com/questions/669452/is-double-square-brackets-preferable-over-single-square-brackets-in-ba
https://serverfault.com/questions/52034/what-is-the-difference-between-double-and-single-square-brackets-in-bash
tldr;
`[ ]` is a POSIX shell expression, it is more portable (runs on more shells and operating systems), but can have unexpected consequences. 
`[[ ]]` is an enhanced Bash expression that is a bit more consistent.

Comment (arthurattwell) [reply test.sh:91]: Thanks, @bertuss, super useful tip. In this case I'm going to leave single `[` for now, given that the script is specifically for the Codeship environment and we're short of time to re-test with double `[[`.
